addons/project/project.py

```diff
@@ -615,48 +581,31 @@ class task(osv.osv):
     _mail_post_access = 'read'
 
     def _get_default_partner(self, cr, uid, context=None):
-        project_id = self._get_default_project_id(cr, uid, context)
-        if project_id:
-            project = self.pool.get('project.project').browse(cr, uid, project_id, context=context)
+        if context is None:
+            context = {}
+        if context.get('default_project_id'):
```

Comment:
If you simply want to check that key `default_project_id` is in dictionary `context`, you can use `in` operator; `get` method reads the value associated to the key.
```python
if 'default_project_id' in context:
```